### PR TITLE
fix(api): container secrets pickup correct environment values

### PR DESCRIPTION
**Pull request by sundersc**
#### Description of changes

When using REST api (containers) in multi-env, push is picking up the previously pushed env secrets instead of the current active environment. This is causing issues with CI/CD setup.

#### Issue #, if available

https://github.com/aws-amplify/amplify-cli/issues/8868

#### Description of how you validated changes
- Manual test
- e2e test in local

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/resources/api-container/express/secret-password-index.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+const checkAuthRules = (req, res, next) => {
+    const jwt = req.header("Authorization") || "";
+
+    const [, jwtBody] = jwt.split(".");
+
+    const obj = JSON.parse(
+        jwtBody ? Buffer.from(jwtBody, "base64").toString("utf-8") : "{}"
+    );
+
+    //Customer can perform logic on JWT body
```
Comment (lazpavel) [inline packages/amplify-e2e-tests/resources/api-container/express/secret-password-index.js:32]: should this be cleaned up?

**Review by lazpavel — APPROVED** [review]
please fix the unused variables warnings, not familiar with the area but code changes look to me with a couple of nits.

File: packages/amplify-e2e-tests/resources/api-container/secrets/.secret-password [hunk 2]
```diff
@@ -0,0 +1 @@
+TESTPASSWORD
```
[the remark this answers is not in the document]
Comment (edwardfoyle) [reply packages/amplify-e2e-tests/resources/api-container/secrets/.secret-password:1]: also nit, can you name the value something a bit more specific like `CONTAINER_SECRETS_TEST_PASSWORD`. This makes grepping for usage from the test back to here more reliable

File: packages/amplify-e2e-core/src/init/amplifyPush.ts [hunk 3]
```diff
@@ -147,6 +147,31 @@ export function amplifyPushWithoutCodegen(
   });
 }
 
+export function amplifyPushSecretsWithoutCodegen(
+  cwd: string,
+  testingWithLatestCodebase: boolean = false,
+  allowDestructiveUpdates: boolean = false,
```
Comment (edwardfoyle) [inline packages/amplify-e2e-core/src/init/amplifyPush.ts:153]: I don't think this is needed here

File: packages/amplify-e2e-tests/resources/api-container/express/secret-password-index.js (hunk 1 above)
Comment (sundersc) [reply packages/amplify-e2e-tests/resources/api-container/express/secret-password-index.js:32]: Removed the unused code.

File: packages/amplify-e2e-tests/resources/api-container/secrets/.secret-password (hunk 2 above)
[the remark this answers is not in the document]
Comment (sundersc) [reply packages/amplify-e2e-tests/resources/api-container/secrets/.secret-password:1]: update the value.

File: packages/amplify-e2e-core/src/init/amplifyPush.ts (hunk 3 above)
Comment (sundersc) [reply packages/amplify-e2e-core/src/init/amplifyPush.ts:153]: removed `allowDestructiveUpdates` as this is not used with GraphQL api.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/utils/containers/set-existing-secret-arns.ts [hunk 4]
```diff
@@ -38,5 +40,16 @@ export const setExistingSecretArns = (secretsMap: Map<string, string>, cfnObj: $
     .flat(1) // merge nested secrets array into one array
     .filter(secretDef => !!secretDef?.Name) // make sure the name is defined
     .filter(secretDef => !!secretDef.ValueFrom) // make sure the arn is defined
-    .forEach(secretDef => secretsMap.set(secretDef.Name, secretDef.ValueFrom)); // add it to the secretsMap map
+    .forEach(secretDef => {
+      if (typeof secretDef.ValueFrom === 'object' && secretDef.ValueFrom['Fn::Join']) {
+        const [delimiter, values] = secretDef.ValueFrom['Fn::Join'];
```
[the remark this answers is not in the document]
Comment (sundersc) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/utils/containers/set-existing-secret-arns.ts:45]: We are reading value from the cfntemplate json and converting back to cdk code here. I can't find a helper method to do this so parsing it manually here.

[the remark this answers is not in the document]
Comment (sundersc) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/utils/containers/set-existing-secret-arns.ts:45]: I'm running e2e tests in local now, will confirm once it is complete.

**Review by edwardfoyle — APPROVED** [review]

**Comment by mattfysh** [comment]
Hi all, just reviewed this PR and ... out of curiosity, how many people in the Amplify team are full across the Serverless Containers implementation?

It does feel like the area of Amplify where it's most difficult to find support. In terms of future plans for SC, will there be a ramping up of support coming this year, or will the Amplify team perhaps decide to drop Containers entirely?

**Pull request merged**